toml/toml.go

```diff
@@ -42,31 +43,54 @@ func (d Duration) MarshalText() (text []byte, err error) {
 }
 
 // Size represents a TOML parseable file size.
-// Users can specify size using "m" for megabytes and "g" for gigabytes.
-type Size int
+// Users can specify size using "k" or "K" for kibibytes, "m" or "M" for mebibytes,
+// and "g" or "G" for gibibytes. If a size suffix isn't specified then bytes are assumed.
+type Size uint64
 
 // UnmarshalText parses a byte size from text.
 func (s *Size) UnmarshalText(text []byte) error {
+	if len(text) == 0 {
+		return fmt.Errorf("size was empty")
+	}
+
+	// The multiplier defaults to 1 in case the size has
+	// no suffix (and is then just raw bytes)
+	mult := int64(1)
+
+	// Preserve the original text for error messages
+	sizeText := text
+
+	// Parse unit of measure
+	suffix := text[len(sizeText)-1]
+	if !unicode.IsDigit(rune(suffix)) {
+		switch suffix {
+		case 'k', 'K':
+			mult = 1 << 10 // KiB
+		case 'm', 'M':
+			mult = 1 << 20 // MiB
+		case 'g', 'G':
+			mult = 1 << 30 // GiB
+		default:
+			return fmt.Errorf("unknown size suffix: %c", suffix)
+		}
+		sizeText = sizeText[:len(sizeText)-1]
+	}
+
 	// Parse numeric portion of value.
-	length := len(string(text))
-	size, err := strconv.ParseInt(string(text[:length-1]), 10, 64)
+	size, err := strconv.ParseInt(string(sizeText), 10, 64)
 	if err != nil {
-		return err
+		return fmt.Errorf("invalid size: %s", string(text))
 	}
 
-	// Parse unit of measure ("m", "g", etc).
-	switch suffix := text[len(text)-1]; suffix {
-	case 'm':
-		size *= 1 << 20 // MB
-	case 'g':
-		size *= 1 << 30 // GB
-	default:
-		return fmt.Errorf("unknown size suffix: %c", suffix)
+	if maxInt/mult < size {
+		return fmt.Errorf("size would overflow the max size (%d) of an int: %s", maxInt, string(text))
```

Comment:
This should say `a uint` because now it's an unsigned integer. I don't think we need to specify it is a uint64. Can you also use `math.MaxUint64` for the size check? It would make it so `maxInt` isn't needed anymore.